### PR TITLE
Reduce diff between upstream pack/unpack and in-tree ones

### PR DESCRIPTION
**Pull request by chelini**
Use `DenseI64ArrayAttr` and reduce diff with upstreamed pack and unpack operations.

### CODE REVIEW

File: lib/TPP/Dialect/LinalgX/LinalgXOps.cpp [hunk 1]
```diff
@@ -128,43 +128,43 @@ LogicalResult UnPackOp::canonicalize(UnPackOp unpackOp,
...
+            innerTiles, builder.getDenseI64ArrayAttr(staticInnerTiles),
+            /*padding_value=*/nullptr);
+    else
+      build(builder, result, TypeRange{}, input, output,
+            builder.getDenseI64ArrayAttr(outerDimPerm),
+            builder.getDenseI64ArrayAttr(innerDimPos), innerTiles,
+            builder.getDenseI64ArrayAttr(staticInnerTiles),
+            /*padding_value=*/{});
+
+  } else {
```
Comment (rengolin) [inline lib/TPP/Dialect/LinalgX/LinalgXOps.cpp:155]: assert(tensor)?

File: lib/TPP/Dialect/LinalgX/LinalgXOps.cpp [hunk 2]
```diff
@@ -176,30 +176,30 @@ void UnPackOp::build(OpBuilder &builder, OperationState &result, Value input,
...
-  SmallVector<int64_t> staticInnerTiles;
-  dispatchIndexOpFoldResults(tiles, innerTiles, staticInnerTiles,
-                             ShapedType::kDynamic);
-  if (!outerDimPerm.size())
-    build(builder, result, llvm::None, input, output, /*outerDimPerm=*/{},
-          innerDimsPos, innerTiles, builder.getI64ArrayAttr(staticInnerTiles));
-  else
-    build(builder, result, llvm::None, input, output, outerDimPerm,
-          innerDimsPos, innerTiles, builder.getI64ArrayAttr(staticInnerTiles));
+  } else {
```
Comment (rengolin) [inline lib/TPP/Dialect/LinalgX/LinalgXOps.cpp:191]: ditto

File: test/TPP/transform/transform-convolutions.mlir [hunk 3]
```diff
@@ -285,3 +285,145 @@ transform.sequence failures(propagate) {
     %9 = transform.structured.interchange %8 { iterator_interchange = [0, 1, 4, 2, 3, 5] }
     transform.structured.map_to_brgemm %9
 }
+
```
Comment (rengolin) [inline test/TPP/transform/transform-convolutions.mlir:288]: please, drop this change. This should be attached to #190 and done in one go, not half-way on an uinrelated PR.

**Comment by rengolin** [comment]
The failure is probably because you need to rebase

**Comment by chelini** [comment]
> Another change is the tensor return type / output,. which is the source of the #45 problem. Can you fix that too? Should be quick and if if fixes the bug, win win!



> sor return type / output,. w

What exactly should I fix it?

**Comment by rengolin** [comment]
The type in the assembly format to look like this: https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/Tensor/IR/TensorOps.td#L1754

**Review by rengolin — APPROVED** [review]
As discussed offline, the change in the return type is due to the op here supporting both tensors and memrefs, and we can't change that now because the upstream ones only support tensors.

This can be a change for another PR. LGTM, thanks!

**Pull request merged**